src/docs/stan-reference/programming.tex

```diff
@@ -10635,10 +10633,176 @@ \subsection{Examples: Fits in Stan}
+
+Because Stan's algorithms (other than Riemannian Hamiltonian Monte
+Carlo) rely on step-based gradient-based approximations of the density
+(or penalized maximum likelihood) being fitted, posterior curvature
+not captured by this first-order appoximation plays a central role in
+determining the statistical efficiency of Stan's algorithms.
+
+A second-order approximation to curvature is provided by the
+Hessian, the matrix of second derivatives of the log density $\log
+p(\theta)$ with respect to the parameter vector $\theta$, defined
+as
+\[
+H(\theta) = \nabla \, \nabla \, \log p(\theta | y),
+\]
+so that
+\[
+H_{i, j}(\theta) = \frac{\partial^2 \log p(\theta | y)}
+                {\partial \theta_i \ \partial \theta_j}.
+\]
+For pure penalized maximum likelihood problems, the posterior log
+density $\log p(\theta | y)$ is replaced by the penalized likelihood
+function $\mathcal{L}(\theta) = \log p(y | \theta) - \lambda(\theta)$.
+
+\subsection{Condition Number and Adaptation}
+
+A good gauge of how difficult a problem the curvature presents is
+given by the condition number of the Hessian matrix $H$, which is the
+ratio of the largest to the smallest eigenvalue of $H$ (assuming the
+Hessian is positive definite).  This essentially measures the
+difference between the flattest direction of movement and the most
+curved.  Typically, the step size of a gradient-based algorithm is
+bounded by the most sharply curved direction.  With better conditioned
+log densities or penalized likelihood functions, it is easier for
+Stan's adaptation, especially the diagonal adaptations that are used
+as defaults.
+
+\subsection{Unit Scales without Correlation}
+
+Ideally, all parameters should be programmed so that they have unit
+scale and so that posterior correlation is reduced; together, these
+properties mean that there is no rotation or scaling required for
+optimal performance of Stan's algorithms.  For Hamiltonian Monte
+Carlo, this implies a unit mass matrix, which requires no adaptation
+as it is where the algorithm initializes.  Riemannian Hamiltonian
+Monte Carlo performs this conditioning on the fly at every step, but
+such conditioning is very expensive computationally.
+
+\subsection{Varying Curvature}
+
+In all but very simple models (such as multivariate normals), the
+Hessian will vary as $\theta$ varies.  As an example, see the
+illustration of Neal's funnel example in \reffigure{funnel}.  The more
+the curvature varies, the harder it is for all of the algorithms with
+fixed adaptation parameters (that is, everything but Riemannian
+Hamiltonian Monte Carlo) to find adaptations that cover the entire
+density well.  Many of the variable transforms proposed are aimed at
+improving the conditioning of the Hessian and/or making it more
+consistent across the relevant portions of the density (or penalized
+maximum likelihood function) being fit.
+
+For all of Stan's algorithm, the curvature along the path from the
```

Comment:
On Wed, Aug 31, 2016 at 1:27 PM, Bob Carpenter <notifications@github.com>
wrote:
>
> Let me know when you're done and I can do a final update. Otherwise, I'll
> just keep hammering our servers.
>

Haven't found anything else, so I think fine to do final update